eclipse/tern.eclipse.ide.server.nodejs.core/src/tern/eclipse/ide/server/nodejs/core/NodejsDebuggersManager.java

```diff
@@ -0,0 +1,47 @@
+/**
+ *  Copyright (c) 2015 Genuitec LLC.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Piotr Tomiak <piotr@genuitec.com> - initial API and implementation
+ */
+package tern.eclipse.ide.server.nodejs.core;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import tern.eclipse.ide.server.nodejs.internal.core.debugging.webclipse.WebclipseDebugger;
+
+public class NodejsDebuggersManager {
+
+	private static Map<String, INodejsDebugger> debuggers = new LinkedHashMap<String, INodejsDebugger>();
+
+	static {
+		// Add supported debuggers
+		addDebugger(new WebclipseDebugger());
```

Comment:
My big problem is that Webclipse will be available in the combo although users could not have this installed debugger. IMHO I think an extension point should be better. I understand your problem that you don't want to have dependenies to tern.java inside Webclipse, so here my suggestion : 

 * create an extension point for tern debugger
 * create a new folder https://github.com/angelozerr/tern.java/tree/master/eclipse/debugger
 * create a new project tern.eclipse.ide.debugger.webclipse which will host WebclipseDebugger

We could have too a new project  tern.eclipse.ide.debugger.core which will provide some abstract classes with launch, but for the moment we have one debugger.

After that we will able to create a new features debugger, and user will able to choose the debugger to install.

What do you think about that?